app/models/concerns/account_finder_concern.rb

```diff
@@ -17,10 +17,49 @@ def find_local(username)
     end
 
     def find_remote(username, domain)
+      return AccountByIdFinder.new(username[4..-1], :mon_id).account if domain.nil? && username.match?(/\Amon_/i)
+      return AccountByIdFinder.new(username[6..-1], :route_no).account if domain.nil? && username.match?(/\Aroute_/i)
       AccountFinder.new(username, domain).account
     end
   end
 
+  class AccountByIdFinder
+    attr_reader :the_id, :id_name
+
+    def initialize(the_id, id_name)
+      @the_id = the_id
+      @id_name = id_name
+    end
+
+    def account
+      scoped_accounts.order(id: :asc).take
+    end
+
+    private
+
+    def scoped_accounts
+      Account.unscoped.tap do |scope|
+        scope.merge! with_id
+        scope.merge! matching_id
+        scope.merge! matching_domain
+      end
+    end
+
+    def with_id
+      search = Hash.new
+      search[id_name] = 0
+      Account.where.not(search)
+    end
+
+    def matching_id
+      Account.where(Account.arel_table[id_name].eq the_id.to_i)
```

Comment:
Isn't this the same as `Account.where(id_name => the_id.to_i)`?

Reply:
*sigh* i have no idea why we're using AREL tables here but since that's how the *other* account finder worked i tried to match the same setup

feel free to change it if it doesn't make a difference though